vochain/scrutinizer/scrutinizer.go

```
@@ -97,33 +94,91 @@ func NewScrutinizer(dbPath string, app *vochain.BaseApplication) (*Scrutinizer,
+// If no CountStore model is stored for the type, it counts all db entries of that type.
+func (s *Scrutinizer) retrieveCounts() (*indexertypes.CountStore,
+	*indexertypes.CountStore, *indexertypes.CountStore, *indexertypes.CountStore) {
+	var err error
+	txCountStore := new(indexertypes.CountStore)
+	if err = s.db.Get(indexertypes.CountStore_Transactions, txCountStore); err != nil {
+		log.Warnf("could not get the transaction count: %v", err)
+		count, err := s.db.Count(&indexertypes.TxReference{}, &badgerhold.Query{})
+		if err != nil {
+			log.Warnf("could not count total transactions: %v", err)
+		} else {
+			// Store new countStore value
+			txCountStore.Count = uint64(count)
+			txCountStore.Type = indexertypes.CountStore_Transactions
+			if err := s.db.Upsert(txCountStore.Type, txCountStore); err != nil {
+				log.Warnf("could not store transaction count: %v", err)
+			}
+		}
+	}
+	envelopeCountStore := new(indexertypes.CountStore)
+	if err = s.db.Get(indexertypes.CountStore_Envelopes, envelopeCountStore); err != nil {
+		log.Warnf("could not get the envelope count: %v", err)
+		count, err := s.db.Count(&indexertypes.VoteReference{}, &badgerhold.Query{})
+		if err != nil {
+			log.Warnf("could not count total envelopes: %v", err)
+		} else {
+			// Store new countStore value
+			envelopeCountStore.Count = uint64(count)
+			envelopeCountStore.Type = indexertypes.CountStore_Envelopes
+			if err := s.db.Upsert(envelopeCountStore.Type, envelopeCountStore); err != nil {
+				log.Warnf("could not store envelope count: %v", err)
+			}
+		}
+	}
+	processCountStore := new(indexertypes.CountStore)
+	if err = s.db.Get(indexertypes.CountStore_Processes, processCountStore); err != nil {
+		log.Warnf("could not get the process count: %v", err)
+		count, err := s.db.Count(&indexertypes.Process{}, &badgerhold.Query{})
+		if err != nil {
+			log.Warnf("could not count total processes: %v", err)
```

That should either return an error or end up in a Fatal